### PR TITLE
refactor: minor changes:

### PR DESCRIPTION
**Pull request by jhz-dev**
- Changing the nodejs version to 22.4.1
- Created a dedicated userController to separate the logic from the router

### CODE REVIEW

**Comment by edwinhern** [comment]
Could you fix merge conflicts, thank you!

File: src/api/user/userController.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import type { Request, RequestHandler, Response } from "express";
+
+import { userServiceInstance } from "@/api/user/userService";
```
Comment (edwinhern) [inline src/api/user/userController.ts:3]: Could you update to 'userService'? similar to the userController class. 

Im liking the pascal case for the classes. 😀

**Review by edwinhern — CHANGES_REQUESTED** [review]
LGTM! Just the small change with service class and merge conflict fix

**Review by edwinhern — APPROVED** [review]

**Review by edwinhern — CHANGES_REQUESTED** [review]
Could you update the `userService` test file. We also need to update the Class initialization name

**Comment by jhz-dev** [comment]
> Could you update the `userService` test file. We also need to update the Class initialization name
> 
> Here's the path: `src/api/user/__tests__/userService.test.ts `

Sure, I will take a look. Thanks

**Comment by jhz-dev** [comment]
@edwinhern the PR is ready for review now. Let me know if something else is needed. Thanks

**Review by edwinhern — APPROVED** [review]

**Pull request merged**